system-addon/lib/HighlightsFeed.jsm

```diff
@@ -49,6 +52,22 @@ this.HighlightsFeed = class HighlightsFeed {
 
   uninit() {
     SectionsManager.disableSection(SECTION_ID);
+    PageThumbs.removeExpirationFilter(this);
+  }
+
+  filterForThumbnailExpiration(callback) {
+    const sectionIndex = SectionsManager.sections.get(SECTION_ID).order;
+    if (this.store.getState().Sections[sectionIndex]) {
+      const {initialized, rows} = this.store.getState().Sections[sectionIndex];
```

Comment:
This unnecessarily calls `getState()` twice in the common case. Probably just convert these 11 lines to:
```js
const state = this.store.getState().Sections[sectionIndex];
callback(state && state.initialized ? state.rows.map(({url}) => url) : []);
```